activejob/lib/active_job/core.rb

```diff
@@ -20,6 +20,9 @@ module Core
 
       # ID optionally provided by adapter
       attr_accessor :provider_job_id
+
+      # I18n.locale to be used during the job.
+      attr_accessor :locale
```

Comment:
Well, it'll be tough to fix until another release. To me locale sounds like the most appropriate name.